Composer/packages/client/src/pages/botProject/adapters/ABSChannels.tsx

```diff
@@ -321,16 +335,27 @@ export const ABSChannels: React.FC<RuntimeSettingsProps> = (props) => {
   };
 
   const hasAuth = async () => {
-    let newtoken = '';
-    if (userShouldProvideTokens()) {
-      if (isShowAuthDialog(false)) {
-        setShowAuthDialog(true);
+    if (currentResource) {
+      let newtoken = '';
+      if (userShouldProvideTokens()) {
+        if (isShowAuthDialog(false)) {
+          setShowAuthDialog(true);
+        }
+        newtoken = getTokenFromCache('accessToken');
+      } else {
+        try {
+          if (currentResource.tenantId) {
+            newtoken = await AuthClient.getARMTokenForTenant(currentResource.tenantId);
+          } else {
+            newtoken = await AuthClient.getAccessToken(armScopes);
+          }
+        } catch (error) {
+          setErrorMessage(error.message || error.toString());
+          setCurrentResource(undefined);
+        }
       }
-      newtoken = getTokenFromCache('accessToken');
-    } else {
-      newtoken = await AuthClient.getAccessToken(armScopes);
+      setToken(newtoken);
     }
-    setToken(newtoken);
```

Comment:
We should consolidate the auth code into a single method that can be called here and above.

Reply:
done